bazel_tools/rules_nodejs_windows_mkdir.patch

```diff
@@ -0,0 +1,31 @@
+diff --git a/internal/npm_install/npm_install.bzl b/internal/npm_install/npm_install.bzl
+index 3c07b3a0..a7b77d9a 100644
+--- a/internal/npm_install/npm_install.bzl
++++ b/internal/npm_install/npm_install.bzl
+@@ -335,8 +335,9 @@ def _copy_file(repository_ctx, f):
+     to_segments = to.split("/")
+     if len(to_segments) > 1:
+         dirname = "/".join(to_segments[:-1])
++        args = ["mkdir", "-p", dirname] if not is_windows_os(repository_ctx) else ["cmd", "/c", "if not exist {dir} mkdir {dir}".format(dir = dirname)]
+         result = repository_ctx.execute(
+-            ["mkdir", "-p", dirname],
++            args,
+             quiet = repository_ctx.attr.quiet,
+         )
+         if result.return_code:
+@@ -344,12 +345,13 @@ def _copy_file(repository_ctx, f):
+ 
+     # copy the file; don't use the repository_ctx.template trick with empty substitution as this
+     # does not copy over binary files properly
++    cp_args = ["cp", "-f", repository_ctx.path(f), to] if not is_windows_os(repository_ctx) else ["cmd", "/c", "copy /y {} {}".format("\\".join(str(repository_ctx.path(f)).split("/")), "\\".join(to_segments))]
```

Comment:
@aherrmann-da do you have a better idea for handling this? `ctx.file` doesn’t work since you either mess up executable bits or break binary files. Just `cp` doesn’t work because it might not be in PATH and this is a repository rule so we cannot get it from the posix toolchain. Just `cmd /c copy` does not work because it does not like forward slashes.
So in the end I ended up with this but it seems terrible.

Reply:
Unfortunately, I don't really have a better idea. As you describe all these other methods have their respective limitations.

Perhaps [`string.replace`](https://docs.bazel.build/versions/master/skylark/lib/string.html#replace) is also a bit less noisy than the split/join, but, I have no strong preference there.